charger/ocpp.go

```diff
@@ -347,38 +365,54 @@ func (c *OCPP) setChargingProfile(connectorId int, profile *types.ChargingProfil
 	return c.wait(err, rc)
 }
 
-// updatePeriod sets a single charging schedule period with given current and phases
-func (c *OCPP) updatePeriod(current float64, phases int) error {
+// updatePeriod sets a single charging schedule period with given current
+func (c *OCPP) updatePeriod(current float64) error {
 	// current period can only be updated if transaction is active
 	if enabled, err := c.Enabled(); err != nil || !enabled {
 		return err
 	}
 
 	current = math.Trunc(10*current) / 10
 
-	err := c.setChargingProfile(c.connector, getTxChargingProfile(current, phases))
+	err := c.setChargingProfile(c.connector, c.getTxChargingProfile(current))
 	if err != nil {
 		err = fmt.Errorf("set charging profile: %w", err)
 	}
 
 	return err
 }
 
-func getTxChargingProfile(current float64, phases int) *types.ChargingProfile {
-	period := types.NewChargingSchedulePeriod(0, current)
+func (c *OCPP) getTxChargingProfile(current float64) *types.ChargingProfile {
+	var phases = c.phases
+	if phases == 0 {
```

Comment:
Separeter PR

Reply:
Für alles, was mit der Umwandlung in die Methode zu tun hat? Will do.

Reply:
Können wir gerne maxhen, ist aber ein separater pr